projects/packages/forms/src/class-wpcom-rest-api-v2-endpoint-forms.php

```diff
@@ -0,0 +1,167 @@
+						'type'     => 'integer',
+						'required' => false,
+						'minimum'  => 1,
+					),
+					'search'  => array(
+						'type'     => 'string',
+						'required' => false,
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Returns Jetpack Forms responses.
+	 *
+	 * @param WP_REST_Request $request The request sent to the WP REST API.
+	 *
+	 * @return WP_REST_Response A response object containing Jetpack Forms responses.
+	 */
+	public function get_responses( $request ) {
+		$args = array(
+			'post_type'   => 'feedback',
+			'post_status' => array( 'publish', 'draft' ),
+		);
+
+		if ( isset( $request['form_id'] ) ) {
+			$args['post_parent'] = $request['form_id'];
+		}
+
+		if ( isset( $request['limit'] ) ) {
+			$args['posts_per_page'] = $request['limit'];
+		}
+
+		if ( isset( $request['offset'] ) ) {
+			$args['offset'] = $request['offset'];
+		}
+
+		if ( isset( $request['search'] ) ) {
+			$args['s'] = $request['search'];
+		}
+
+		$query = new \WP_Query( $args );
+
+		$responses = array_map(
+			function ( $response ) {
+				$data = \Automattic\Jetpack\Forms\ContactForm\Contact_Form_Plugin::parse_fields_from_content( $response->ID );
+
+				return array(
+					'id'                      => $response->ID,
+					'uid'                     => $data['all_fields']['feedback_id'],
+					'date'                    => get_the_date( 'c', $response ),
+					'author_name'             => $data['_feedback_author'],
+					'author_email'            => $data['_feedback_author_email'],
+					'author_url'              => $data['_feedback_author_url'],
+					'author_avatar'           => empty( $data['_feedback_author_email'] ) ? '' : get_avatar_url( $data['_feedback_author_email'] ),
+					'email_marketing_consent' => $data['all_fields']['email_marketing_consent'],
+					'ip'                      => $data['_feedback_ip'],
+					'entry_title'             => $data['all_fields']['entry_title'],
+					'entry_permalink'         => $data['all_fields']['entry_permalink'],
+					'subject'                 => $data['_feedback_subject'],
+					'fields'                  => array_diff_key(
+						$data['all_fields'],
+						array(
+							'email_marketing_consent' => '',
+							'entry_title'             => '',
+							'entry_permalink'         => '',
+							'feedback_id'             => '',
+						)
+					),
+				);
+			},
+			$query->posts
+		);
+
+		return rest_ensure_response(
+			array(
+				'responses' => $responses,
+				'total'     => $query->found_posts,
+			)
+		);
+	}
+
+	/**
+	 * Verifies that the current user has the requird capability for viewing form responses.
+	 *
+	 * @return true|WP_Error Returns true if the user has the required capability, else a WP_Error object.
+	 */
+	public function get_responses_permission_check() {
+		$site_id = Manager::get_site_id();
+		if ( is_wp_error( $site_id ) ) {
+			return $site_id;
+		}
+
+		if ( ! current_user_can( 'manage_options' ) || ! is_user_member_of_blog( get_current_user_id(), $site_id ) ) {
```

Comment:
With `current_user_can( 'manage_options' )` in place, then `is_user_member_of_blog( get_current_user_id(), $site_id )` becomes redundant, right?

In that case we don't need the `$site_id` as well I guess.

Reply:
User might be member of blog but not have management capabilities? Is that not a possible scenario?

Reply:
No worries, doesn't hurt anyway :)

Reply:
I followed up on this conversation in #35571 after discussion in p1707435111778979-slack-C05PV073SG3